client/blocks/shortcode/index.js

```diff
@@ -0,0 +1,55 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryShortcode from 'components/data/query-shortcode';
+import { getShortcode } from 'state/shortcodes/selectors';
+
+/**
+ * Local dependencies
+ */
+import ShortcodeFrame from './frame';
+
+const Shortcode = props => {
+	const { siteId, className, children, filterRenderResult, shortcode } = props;
+	const classes = classNames( 'shortcode', className );
+	let filteredShortcode = {};
+	if ( shortcode ) {
+		shortcode.body = shortcode.result;
+		filteredShortcode = filterRenderResult( omit( shortcode, 'shortcode' ) );
+	}
+	return (
+		<div>
+			<QueryShortcode siteId={ siteId } shortcode={ children } />
+			<ShortcodeFrame
+				{ ...omit( props, 'siteId', 'filterRenderResult', 'shortcode', 'dispatch' ) }
```

Comment:
These `omit` calls make it very hard to figure out what the component's API actually is and what props are possible.

But they are used everywhere in the related components (`GalleryShortcode` et al) and it will be hard to get rid of them. Topic for a standalone PR rather than additional change in this one.